### PR TITLE
use FontConfig_jll instead of BinaryProvider

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

**Comment by KristofferC** [comment]
These repo doesn't have many tests so would be good to test that the reverse dependencies still work properly.

**Comment by KristofferC** [comment]
Hm, @SimonDanisch, any idea about:

```
ERROR: LoadError: Fontconfig was unable to match font Fontconfig.Pattern("Helvetica-10:familylang=en,en-us:stylelang=en,en-us:fullnamelang=en,en-us:slant=0:weight=80:width=100:pixelsize=10.4167:hintstyle=3:hinting=True:verticallayout=False:autohint=False:globaladvance=True:dpi=75:scale=1:lang=en:fontversion=2147483647:embeddedbitmap=True:decorative=False:namelang=en:prgname=julia:symbol=False:variable=False")
``

**Comment by SimonDanisch** [comment]
Uhm... I guess that font doesn't exist on macos? :D
I'm not sure, if anyone actually uses this package... It wasn't working for quite a while, and nobody complained... And I haven't had the time to make it stable enough to integrate it into FreeTypeAbstraction ;)

**Comment by KristofferC** [comment]
Looks like there are some that depend on it at least: https://github.com/JuliaRegistries/General/search?q=Fontconfig&unscoped_q=Fontconfig

File: src/Fontconfig.jl [hunk 1]
```diff
@@ -1,25 +1,17 @@
...
-import Base.Sys
 using Printf
+using Fontconfig_jll
 
 export format, match, list
 
 function __init__()
-    ENV["FONTCONFIG_FILE"] = joinpath(dirname(jl_libfontconfig), "..", "etc", "fonts", "fonts.conf")
-    ccall((:FcInit, jl_libfontconfig), UInt8, ())
+    ENV["FONTCONFIG_FILE"] = joinpath(dirname(libfontconfig), "..", "etc", "fonts", "fonts.conf")
```
Comment (giordano) [inline src/Fontconfig.jl]: `fonts_conf` is [one of the products](https://github.com/JuliaPackaging/Yggdrasil/blob/930b004d197192c527c259f52ef4475ae2f848b4/F/Fontconfig/build_tarballs.jl#L68) exactly to make this easier:
```suggestion
    ENV["FONTCONFIG_FILE"] = fonts_conf
```
:wink:

**Comment by KristofferC** [comment]
Passes now :) cc @SimonDanisch

**Comment by SimonDanisch** [comment]
Hm, still can't get this to work on my ubuntu 19.10, so guess there is still something not right - not sure if that should be a blocker, considering that its not necessarily a regression... After all, this branch even seems to work on windows :D

**Comment by KristofferC** [comment]
I'm not getting blocked on anything by this PR so no rush from my end. Just wanted to get rid of some BP :P

**Comment by SimonDanisch** [comment]
Lets just merge it :D

**Comment by asinghvi17** [comment]
Bump @SimonDanisch

**Comment by ViralBShah** [comment]
Bump. Seems very close. Can we get this merged?

**Pull request merged**